axelor-account/src/main/java/com/axelor/apps/account/web/FixedAssetController.java

```diff
@@ -125,4 +205,62 @@ public void massValidation(ActionRequest request, ActionResponse response) {
       TraceBackService.trace(response, e);
     }
   }
+
+  public void computeFirstDepreciationDate(ActionRequest request, ActionResponse response) {
+
+    try {
+      FixedAsset fixedAsset = request.getContext().asType(FixedAsset.class);
+      Beans.get(FixedAssetService.class).computeFirstDepreciationDate(fixedAsset);
+
+      response.setValue("firstDepreciationDate", fixedAsset.getFirstDepreciationDate());
+
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void updateDepreciation(ActionRequest request, ActionResponse response) {
+    try {
+      FixedAsset fixedAsset = request.getContext().asType(FixedAsset.class);
+      Beans.get(FixedAssetService.class).updateDepreciation(fixedAsset);
+      response.setValue("fixedAssetLineList", fixedAsset.getFixedAssetLineList());
+      response.setValue("fiscalFixedAssetLineList", fixedAsset.getFiscalFixedAssetLineList());
+      response.setValue(
+          "fixedAssetDerogatoryLineList", fixedAsset.getFixedAssetDerogatoryLineList());
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void splitFixedAsset(ActionRequest request, ActionResponse response) {
+    Context context = request.getContext();
+    Long fixedAssetId = Long.valueOf(context.get("_id").toString());
+    FixedAsset fixedAsset = Beans.get(FixedAssetRepository.class).find(fixedAssetId);
+    BigDecimal disposalQty = new BigDecimal(context.get("qty").toString());
+
+    try {
+      if (disposalQty.compareTo(fixedAsset.getQty()) > 0) {
+        throw new AxelorException(
+            TraceBackRepository.CATEGORY_INCONSISTENCY,
+            I18n.get(IExceptionMessage.IMMO_FIXED_ASSET_DISPOSAL_QTY_GREATER_ORIGINAL),
+            fixedAsset.getQty().toString());
+      }
+      FixedAsset createdFixedAsset =
+          Beans.get(FixedAssetService.class)
+              .splitAndSaveFixedAsset(
+                  fixedAsset, disposalQty, LocalDate.now(), fixedAsset.getComments());
```

Comment:
We should use appBaseService.getTodayDate() here

Reply:
done